Decorator.cpp

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <string>
+
+class Burger
+{
+public:
+    virtual ~Burger(){};
+    virtual std::string Serve() = 0;
+    virtual float Price() = 0;
+};
+
+class ClassicBurger : public Burger
+{
+private:
+    float m_price = 10.2;
+public:
```

Comment:
Please add a destructor. `~ClassicBurger() override = default;`
You can take this as a rule: if there are polymorphic classes do not forget to override the sub-class destructor.